app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,51 @@
+class CommentsController < ApplicationController
+  before_action :require_login, only: [:new, :create, :destroy]
+
+  def new
+    @comment=Comment.new
+  end
+
+  def create
+    if params[:question_id]
+      q_or_a = Question.find(params[:question_id])
+    elsif
+      q_or_a = Answer.find(params[:answer_id])
+    end
+
+    comment = q_or_a.comments.find_or_initialize_by(comment_params)
```

Comment:
In a create you wouldn't expecd to find a record, you could could just do q_or_a.comments.build